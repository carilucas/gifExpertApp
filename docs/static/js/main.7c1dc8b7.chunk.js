(this["webpackJsonp04-gif-expert-app"]=this["webpackJsonp04-gif-expert-app"]||[]).push([[0],{17:function(e,t,n){},18:function(e,t,n){"use strict";n.r(t);var a=n(0),r=n(1),c=n(8),i=n.n(c),s=n(2),u=n(10),o=function(e){var t=e.setCategories,n=Object(r.useState)(""),c=Object(s.a)(n,2),i=c[0],o=c[1];return Object(a.jsx)("form",{action:"",onSubmit:function(e){e.preventDefault(),i.trim().length>2&&(t((function(e){return[i].concat(Object(u.a)(e))})),o(""))},children:Object(a.jsx)("input",{type:"text",value:i,onChange:function(e){o(e.target.value)},placeholder:"Ingrese un anime",autoFocus:!0})})},j=n(7),d=n(6),l=n.n(d),p=n(9),m=function(){var e=Object(p.a)(l.a.mark((function e(t){var n,a,r,c;return l.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.prev=0,"w96qRFI3PbHoipqO9YaDTyfOo0SPwNlF",e.next=4,fetch("https://api.giphy.com/v1/gifs/search?q=".concat(encodeURI(t),"&limit=10&api_key=").concat("w96qRFI3PbHoipqO9YaDTyfOo0SPwNlF"));case 4:return n=e.sent,e.next=7,n.json();case 7:return a=e.sent,r=a.data,c=r.map((function(e){return{id:e.id,title:e.title?e.title:"Sin T\xedtulo",url:e.images.downsized_medium.url?e.images.downsized_medium.url:"No existe esta imagen"}})),e.abrupt("return",c);case 13:throw e.prev=13,e.t0=e.catch(0),new Error(e.t0);case 16:case"end":return e.stop()}}),e,null,[[0,13]])})));return function(t){return e.apply(this,arguments)}}(),b=function(e){var t=e.title,n=e.url;return Object(a.jsxs)("div",{className:"animate__animated animate__fadeIn card",children:[Object(a.jsx)("h4",{children:t}),Object(a.jsx)("img",{src:n,alt:t})]})},f=function(e){var t=e.category,n=function(e){var t=Object(r.useState)({data:[],loading:!0}),n=Object(s.a)(t,2),a=n[0],c=n[1];return Object(r.useEffect)((function(){m(e).then((function(e){c({data:e,loading:!1})}))}),[e]),a}(t),c=n.data,i=n.loading;return Object(a.jsxs)(a.Fragment,{children:[i&&Object(a.jsx)("h4",{className:"animate__animated animate__flash",children:"Cargando..."}),Object(a.jsx)("h3",{className:"animate__animated animate__fadeIn",children:t.toUpperCase()}),Object(a.jsx)("div",{className:"card-grid",children:c.map((function(e){return Object(r.createElement)(b,Object(j.a)(Object(j.a)({},e),{},{key:e.id}))}))})]})},O=function(){var e=Object(r.useState)(["naruto"]),t=Object(s.a)(e,2),n=t[0],c=t[1];return Object(a.jsxs)(a.Fragment,{children:[Object(a.jsx)("h2",{children:"GifExpertApp"}),Object(a.jsx)(o,{setCategories:c}),Object(a.jsx)("hr",{}),Object(a.jsx)("ul",{children:n.map((function(e){return Object(a.jsx)(f,{category:e},e)}))})]})};n(17);i.a.render(Object(a.jsx)(O,{}),document.getElementById("root"))}},[[18,1,2]]]);
//# sourceMappingURL=main.7c1dc8b7.chunk.js.map