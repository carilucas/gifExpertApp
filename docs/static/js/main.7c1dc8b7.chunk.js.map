{"version":3,"sources":["components.js/AddCategory.js","helpers/GetGifs.js","components.js/GifGridItem.js","components.js/GifGrid.js","hooks/useFetchGifs.js","GifExpertApp.js","index.js"],"names":["AddCategory","setCategories","useState","inputValue","setInputValue","action","onSubmit","e","preventDefault","trim","length","categories","type","value","onChange","target","placeholder","autoFocus","getGifs","category","a","fetch","encodeURI","respuesta","json","data","gifs","map","img","id","title","url","images","downsized_medium","Error","GifGridItem","className","src","alt","GifGrid","loading","state","setState","useEffect","then","useFetchGifs","toUpperCase","key","GifExpertApp","ReactDOM","render","document","getElementById"],"mappings":"0MAGaA,EAAc,SAAC,GAAqB,IAApBC,EAAmB,EAAnBA,cAAmB,EACTC,mBAAS,IADA,mBACtCC,EADsC,KAC1BC,EAD0B,KAY7C,OACG,sBAAMC,OAAO,GAAGC,SARE,SAACC,GACnBA,EAAEC,iBACEL,EAAWM,OAAOC,OAAO,IAC1BT,GAAc,SAAAU,GAAU,OAAGR,GAAH,mBAAiBQ,OACzCP,EAAc,MAIjB,SACG,uBACGQ,KAAK,OACLC,MAAOV,EACPW,SAfiB,SAAAP,GACvBH,EAAcG,EAAEQ,OAAOF,QAejBG,YAAY,mBACZC,WAAS,O,8BCnBRC,EAAO,uCAAG,WAAMC,GAAN,qBAAAC,EAAA,sEAEF,mCAFE,SAGOC,MAAM,0CAAD,OAA4CC,UAAUH,GAAtD,6BADd,qCAFE,cAGXI,EAHW,gBAIIA,EAAUC,OAJd,uBAIVC,EAJU,EAIVA,KAEDC,EAAOD,EAAKE,KAAI,SAAAC,GAAG,MAAG,CACzBC,GAAGD,EAAIC,GACPC,MAAMF,EAAIE,MAAQF,EAAIE,MAAQ,gBAC9BC,IAAIH,EAAII,OAAOC,iBAAiBF,IAAMH,EAAII,OAAOC,iBAAiBF,IAAM,4BAT1D,kBAWVL,GAXU,wCAaX,IAAIQ,MAAJ,MAbW,0DAAH,sDCDPC,EAAc,SAAC,GAAiB,IAAhBL,EAAe,EAAfA,MAAMC,EAAS,EAATA,IAEhC,OACG,sBAAKK,UAAU,yCAAf,UACG,6BAAKN,IACL,qBAAKO,IAAKN,EAAKO,IAAKR,QCFhBS,EAAU,SAAC,GAAgB,IAAfpB,EAAc,EAAdA,SAAc,ECFX,SAACA,GAAc,IAAD,EACbjB,mBAAS,CAChCuB,KAAK,GACLe,SAAS,IAH2B,mBAChCC,EADgC,KACzBC,EADyB,KAevC,OAVAC,qBAAW,WACRzB,EAAQC,GACPyB,MAAK,SAAAlB,GACHgB,EAAS,CACNjB,KAAKC,EACLc,SAAQ,SAGf,CAACrB,IAEIsB,EDXuBI,CAAa1B,GAA/Ba,EAFwB,EAE7BP,KAAYe,EAFiB,EAEjBA,QACnB,OACG,qCACIA,GAAW,oBAAIJ,UAAU,mCAAd,yBACZ,oBAAIA,UAAU,oCAAd,SAAmDjB,EAAS2B,gBAC5D,qBAAKV,UAAU,YAAf,SAEMJ,EAAOL,KAAI,SAAAC,GAAG,OAAE,wBAAC,EAAD,2BAAiBA,GAAjB,IAAsBmB,IAAKnB,EAAIC,eEVjDmB,EAAe,WAAO,IAAD,EAEK9C,mBAAS,CAAC,WAFf,mBAExBS,EAFwB,KAEZV,EAFY,KAI/B,OACG,qCACG,8CACA,cAAC,EAAD,CAAaA,cAAeA,IAC5B,uBAEA,6BAEGU,EAAWgB,KAAI,SAACR,GACb,OAAM,cAAC,EAAD,CAASA,SAAUA,GAAeA,Y,MCVvD8B,IAASC,OACL,cAAC,EAAD,IACFC,SAASC,eAAe,W","file":"static/js/main.7c1dc8b7.chunk.js","sourcesContent":["import React, { useState } from 'react';\nimport PropTypes from 'prop-types'\n\nexport const AddCategory = ({setCategories}) => {\n   const [inputValue, setInputValue] = useState('');\n   const handleInputChange = e =>{\n      setInputValue(e.target.value);\n   }\n   const handleSubmit = (e)=>{\n      e.preventDefault();\n      if (inputValue.trim().length>2) {\n         setCategories(categories=>[inputValue,...categories]);\n         setInputValue('')\n      }\n   }\n   return (\n      <form action=\"\" onSubmit={handleSubmit}>\n         <input \n            type=\"text\"\n            value={inputValue}\n            onChange={handleInputChange}\n            placeholder=\"Ingrese un anime\"\n            autoFocus\n         />\n      </form>\n   )\n}\n\nAddCategory.propTypes = {\n   setCategories: PropTypes.func.isRequired\n}","\n\n\nexport const getGifs = async(category)=>{\n   try {\n      const apiKey = 'w96qRFI3PbHoipqO9YaDTyfOo0SPwNlF'\n      const respuesta = await fetch(`https://api.giphy.com/v1/gifs/search?q=${ encodeURI(category) }&limit=10&api_key=${apiKey}`);\n      const {data} = await respuesta.json();\n      \n      const gifs = data.map(img=>({\n         id:img.id,\n         title:img.title ? img.title : 'Sin TÃ­tulo',\n         url:img.images.downsized_medium.url ? img.images.downsized_medium.url : 'No existe esta imagen'\n      }));\n      return gifs;\n   } catch (error) {\n      throw new Error(error)\n   }\n}","import React from 'react'\n\nexport const GifGridItem = ({title,url}) => {\n   \n   return (\n      <div className=\"animate__animated animate__fadeIn card\">\n         <h4>{title}</h4>\n         <img src={url} alt={title}/>\n      </div>\n   )\n}\n","import React from 'react'\nimport { useFetchGifs } from '../hooks/useFetchGifs';\nimport { GifGridItem } from './GifGridItem';\n\n\nexport const GifGrid = ({category}) => {\n    \n   const {data:images,loading} = useFetchGifs(category);\n   return(\n      <>\n         {loading && <h4 className=\"animate__animated animate__flash\">Cargando...</h4>}\n         <h3 className=\"animate__animated animate__fadeIn\">{category.toUpperCase()}</h3>\n         <div className=\"card-grid\">\n            {\n               images.map(img=><GifGridItem {...img} key={img.id}/>)\n            }\n         </div>\n      </>\n   )\n}\n","import {useState,useEffect} from 'react';\nimport {getGifs} from '../helpers/GetGifs'\n\nexport const useFetchGifs = (category) => {\n   const [state, setState] = useState({\n      data:[],\n      loading: true\n   })\n   useEffect( ()=>{\n      getGifs(category)\n      .then(gifs=>{\n         setState({\n            data:gifs,\n            loading:false\n         });\n      })\n   },[category])\n   \n   return state;\n}\n","import React , {useState}from 'react'\nimport { AddCategory } from './components.js/AddCategory';\nimport { GifGrid } from './components.js/GifGrid';\n\nexport const GifExpertApp = () => {\n   \n   const [categories, setCategories] = useState(['naruto']);\n   \n   return (\n      <>\n         <h2>GifExpertApp</h2>\n         <AddCategory setCategories={setCategories}/>\n         <hr/>\n         \n         <ul>\n         {\n            categories.map((category) =>{\n               return<GifGrid category={category} key={category}/>;\n            })\n         }\n         </ul>\n         \n      </>\n   )\n};\n","import React from 'react';\nimport ReactDOM from 'react-dom';\nimport { GifExpertApp } from './GifExpertApp';\nimport './index.css';\n\n\n\nReactDOM.render(\n    <GifExpertApp />,\n  document.getElementById('root')\n);\n\n// If you want to start measuring performance in your app, pass a function\n// to log results (for example: reportWebVitals(console.log))\n// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals\n\n"],"sourceRoot":""}