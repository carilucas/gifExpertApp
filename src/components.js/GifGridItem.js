import React from 'react';
import PropTypes from 'prop-types'

export const GifGridItem = ({title,url}) => {
   
   return (
      <div className="animate__animated animate__fadeIn card">
         <h4>{title}</h4>
         <img src={url} alt={title}/>
      </div>
   )
}

GifGridItem.propTypes={
   title:PropTypes.string.isRequired,
   url:PropTypes.string.isRequired,
}