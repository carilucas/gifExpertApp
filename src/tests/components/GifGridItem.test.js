import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components.js/GifGridItem'

describe('Test del GifgridItem', ()=>{
   const title = 'goku';
   const url = 'http://url.com';
   let wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
   beforeEach(()=>{
      wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
   });
   test('Debe mostrar el componente correctamente', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('El tÃ­tulo debe estar en el h4', () => {
      const titleExpect = wrapper.find('h4').text();
      expect(titleExpect).toBe(title);
   });
   
   test('La imagen debe tener la url  y alt', () => {
      const urlExpect = wrapper.find('img');
      
      expect(urlExpect.props().src).toBe(url);
      expect(urlExpect.props().alt).toBe(title);
      
   });
   
   test('El div externo Debe tener la clase animate__fadeIn', () => {
      const div = wrapper.find('div');
      
      expect(div.props().className.includes('animate__fadeIn')).toBe(true)
      
   });
})