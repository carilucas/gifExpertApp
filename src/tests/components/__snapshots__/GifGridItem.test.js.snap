// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del GifgridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="animate__animated animate__fadeIn card"
>
  <h4>
    goku
  </h4>
  <img
    alt="goku"
    src="http://url.com"
  />
</div>
`;
